Kernel/SharedBuffer.cpp

```diff
@@ -67,11 +89,14 @@ void SharedBuffer::deref_for_process(Process& process)
 #ifdef SHARED_BUFFER_DEBUG
                 dbgprintf("Released shared buffer reference on %d of size %d by PID %d\n", m_shared_buffer_id, size(), process.pid());
 #endif
+                sanity_check("deref_for_process");
                 destroy_if_unused();
                 return;
             }
         }
     }
+
+    ASSERT_NOT_REACHED();
 }
 
 void SharedBuffer::disown(pid_t pid)
```

Comment:
Still not right. The disown handling is broken, as it doesn't alter m_total_refs.